utils-kotlin/src/main/kotlin/org/sonarsource/kotlin/externalreport/common/Translator.kt

```diff
@@ -0,0 +1,394 @@
+package org.sonarsource.kotlin.externalreport.common
+
+import java.io.StringWriter
+import java.io.Writer
+import java.util.BitSet
+
+
+object Translator {
+
+    /** The first character of each key in the lookupMap.  */
+    private var prefixSet: BitSet = BitSet()
+
+    /** The length of the shortest key in the lookupMap.  */
+    private var shortest = 0
+
+    /** The length of the longest key in the lookupMap.  */
+    private var longest = 0
+
+    init {
+        var currentShortest = Int.MAX_VALUE
+        var currentLongest = 0
+        for ((key, _) in TO_ESCAPE) {
+            prefixSet.set(key[0].code)
+            val sz = key.length
+            if (sz < currentShortest) {
+                currentShortest = sz
+            }
+            if (sz > currentLongest) {
+                currentLongest = sz
+            }
+        }
+        shortest = currentShortest
+        longest = currentLongest
+    }
+
+    fun escapeHtml4(input: CharSequence): String {
+        val writer = StringWriter(input.length * 2)
+        translate(input, writer)
+        return writer.toString()
+
+    }
+
+    fun translate(input: CharSequence, out: Writer) {
+        var pos = 0
+        val len = input.length
+        while (pos < len) {
+            val consumed = translate(input, pos, out)
+            if (consumed == 0) {
+                // inlined implementation of Character.toChars(Character.codePointAt(input, pos))
+                // avoids allocating temp char arrays and duplicate checks
```

Comment:
I don't quite understand this comment. Why is the `Character.toChars(Character.codePointAt(input, pos))` relevant here?